app/src/main/java/org/fossasia/openevent/general/event/EventViewHolder.kt

```diff
@@ -44,10 +44,14 @@ class EventViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
         itemView.shareFab.setOnClickListener {
             val sendIntent = Intent()
             sendIntent.action = Intent.ACTION_SEND
-
-            sendIntent.putExtra(Intent.EXTRA_TEXT, EventUtils.getSharableInfo(event))
             sendIntent.type = "text/plain"
-            itemView.context.startActivity(Intent.createChooser(sendIntent, "Share Event Details"))
+            sendIntent.putExtra(Intent.EXTRA_TEXT, EventUtils.getSharableInfo(event))
+            val bmpUri = EventUtils.getLocalBitmapUri(itemView.eventImage)
+            if (bmpUri != null) {
+                sendIntent.type = "image/*"
+                sendIntent.putExtra(Intent.EXTRA_STREAM, bmpUri)
+            }
+            itemView.eventImage.context.startActivity(Intent.createChooser(sendIntent, "Share Event Details"))
```

Comment:
Duplicate logic. Create a util function

Reply:
@iamareebjamal sorry didn't get you.

Reply:
This logic is duplicated. Create a util function to reuse code

Reply:
done! please review @iamareebjamal